c_sharp_json_diff/JsonDiff.cs

```diff
@@ -67,71 +70,17 @@ public string NumericDiffAsString(string left, string right)
 
         }
 
-        private JObject CalculateNumericDiff(JObject jObject, bool isAbsoluteDiff)
-        {
-            foreach (KeyValuePair<string, JToken?> prop in jObject)
-            {
-                if (ShouldSkipProperty(prop.Key))
-                {
-                    continue;
-                }
-                if (prop.Value != null)
-                {
-                    if (prop.Value is JArray valuesAsArray)
-                    {
-                        if (valuesAsArray.Count >= 2)
-                        {
-                            if (valuesAsArray[0] != null && valuesAsArray[1] != null)
-                            {
-                                string leftValueAsString = valuesAsArray[0].ToString();
-                                string rightValueAsString = valuesAsArray[1].ToString();
-                                if (int.TryParse(leftValueAsString, out int leftValueAsInt) && int.TryParse(rightValueAsString, out int rightValueAsInt))
-                                {
-                                    int diffValue = isAbsoluteDiff ? Math.Abs(leftValueAsInt - rightValueAsInt) : leftValueAsInt - rightValueAsInt;
-                                    var map = CreateDictionary(leftValueAsInt, rightValueAsInt, diffValue);
-                                    JToken jToken = JToken.FromObject(map);
-                                    prop.Value.Replace(jToken);
-                                }
-                                else if (double.TryParse(leftValueAsString, out double leftValueAsDouble) && double.TryParse(rightValueAsString, out double rightValueAsDouble))
-                                {
-                                    double diffValue = isAbsoluteDiff ? Math.Abs(leftValueAsDouble - rightValueAsDouble) : leftValueAsDouble - rightValueAsDouble;
-                                    var map = CreateDictionary(leftValueAsDouble, rightValueAsDouble, diffValue);
-                                    JToken jToken = JToken.FromObject(map);
-                                    prop.Value.Replace(jToken);
-                                }
-
-                            }
-                        }
-                    }
-                    else
-                    {
-                        if (prop.Value is JObject jsonObject)
-                        {
-                            CalculateNumericDiff(jsonObject, isAbsoluteDiff);
-                        }
-                    }
-                }
-            }
-            return jObject;
-        }
-
         private string JsonDiffBetweenTwo(string left, string right)
         {
             return new JsonDiffPatch(new Options { ArrayDiff = ArrayDiffMode.Efficient }).Diff(left, right);
         }
 
-        private Dictionary<object, object> CreateDictionary(object left, object right, object diff)
+        private void InjectProcessors(Dictionary<string, IJsonProcessor<object>> processors, JObject jObject)
```

Comment:
The order of the processors is important. We expect processors to mutate the ``jObject`` . So we should pass the processors as a list rather than a dictionary. We are only using the value anyway, so a dictionary is not needed here.

Reply:
Refactored to be list